react-native-syrf-client/ios/SyrfClient.swift

```diff
@@ -87,4 +158,114 @@ class SyrfClient: NSObject {
     
 }
 
+// MARK: - Extension for LocationManager Delegate
+extension SyrfClient: LocationDelegate {
+    
+    func locationFailed(_ error: Error) {
+        self.sendEvent(eventName: FAILED_LOCATION_EVENT, data: ["error": error.localizedDescription])
+    }
+    
+    func locationUpdated(_ location: SYRFLocation) {
+        print("location updated: \(location)")
+        self.sendEvent(eventName: UPDATE_LOCATION_EVENT, data: self.getLocationDictionary(location))
+    }
+    
+    func currentLocationUpdated(_ location: SYRFLocation) {
+        print("current location updated: \(location) ")
+        self.sendEvent(eventName: UPDATE_LOCATION_EVENT, data: self.getLocationDictionary(location))
+    }
+}
+
+// MARK: - Extension for PermissionsManager Delegate
+extension SyrfClient: PermissionsDelegate {
+    
+    func authorizationUpdated(_ status: PermissionsAuthorization) {
+        if let callback = self.callbackAuthorization {
+            callback(self.getAuthorizationStatus(status))
+            self.callbackAuthorization = nil
+        }
+    }
+    
+    func accuracyUpdated(_ status: PermissionsAccuracy) {
+        if let callback = self.callbackAccuracy {
+            callback(self.getAccuracyStatus(status))
+            self.callbackAccuracy = nil
+        }
+    }
+}
+
+// MARK: - Extension for RN Module
+extension SyrfClient {
+    
+    func getAuthorizationStatus(_ status: PermissionsAuthorization) -> String {
+        switch status {
+            case .notAvailable:
+                return "notAvailable"
+            case .notDetermined:
+                return "notDetermined"
+            case .notAuthorized:
+                return "notAuthorized"
+            case .authorizedAlways:
+                return "authorizedAlways"
+            case .authorizedWhenInUse:
+                return "authorizedWhenInUse"
+        }
+    }
+    
+    func getAccuracyStatus(_ status: PermissionsAccuracy) -> String {
+        switch status {
+            case .notAvailable:
+                return "notAvailable"
+            case .full:
+                return "full"
+            case .reduced:
+                return "reduced"
+        }
+    }
+    
+    func getActivityType(activity: String) -> CLActivityType {
+        if (activity.lowercased() == "airborne") {
+            return .airborne
+        } else if (activity.lowercased() == "othernavigation") {
+            return .otherNavigation
+        } else if (activity.lowercased() == "automotivenavigation") {
+            return .automotiveNavigation
+        } else if (activity.lowercased() == "fitness") {
+            return .fitness
+        } else {
+            return .other
+        }
+    }
+    
+    func getAccuracyFilter(accuracy: String) -> CLLocationAccuracy {
```

Comment:
We can simplify here the structure like 
if (....) {
    return ....
}
if (....) {
    return ....
}